### PR TITLE
JAMES-2586 - Implement PostgresTableManager

### PR DESCRIPTION
**Pull request by vttranlina**
jira Jira https://issues.apache.org/jira/browse/JAMES-2586

### CODE REVIEW

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTable.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.function.Function;
+
+import org.jooq.DDLQuery;
+import org.jooq.DSLContext;
+
+public class PostgresTable {
+
+    @FunctionalInterface
+    public interface RequireCreateTableStep {
+        PostgresTable creteTableStep(Function<DSLContext, DDLQuery> createTableStep);
```
Comment (quantranhong1999) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTable.java:31]:
```suggestion
        PostgresTable createTableStep(Function<DSLContext, DDLQuery> createTableStep);
```

File: backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+    Function<PostgresModule, PostgresTableManager> tableManagerFactory = module -> new PostgresTableManager(new PostgresExecutor(connectionFactory.create()
+        .map(c -> c)), module);
+
+    @Test
+    void initializeTableShouldSuccessWhenModuleHasSingleTable() {
+        String tableName = "tableName1";
+
+        PostgresTable table = PostgresTable.name(tableName)
+            .creteTableStep(dslContext -> dslContext.createTable(tableName)
```
Comment (quantranhong1999) [inline backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java:86]: What happens if `PostgresTable.name(tableA)` and `dslContext.createTable(tableB)`?

Maybe `dslContext.createTable(tableName)` should be implicitly set following `PostgresTable.name(tableName)` (like Cassandra), no?

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresModule.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        public Builder() {
+            tables = ImmutableList.builder();
+            tableIndexes = ImmutableList.builder();
+        }
+
+        public Builder table(PostgresTable... table) {
+            tables.add(table);
+            return this;
+        }
```
Comment (quantranhong1999) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresModule.java:79]: Q: Should we change `table` -> `addTable`?

Today:
```
builder
  .table(tableA)
  .table(tableB)
```

My impression when use that code is tableB would override `tableA`, and only `tableB` is created.

File: backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java (hunk 2 above)
Comment (vttranlina) [reply backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java:86]: I thought it at first, But I can't not find a way to do it by code
Let me try one more time

**Comment by quantranhong1999** [comment]
Should target [apache:postgresql](https://github.com/apache/james-project/tree/postgresql) instead of master.

**Comment by vttranlina** [comment]
> Should target [apache:postgresql](https://github.com/apache/james-project/tree/postgresql) instead of master.

I created a new pr to `postgresql` branch at https://github.com/apache/james-project/pull/1770
And add a new fixup commit
I will close this pr

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresModule.java (hunk 3 above)
Comment (vttranlina) [reply backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresModule.java]: https://github.com/apache/james-project/pull/1770/commits/8089a3cb1adc20353817e213364a8ff45b9a8559

**Pull request closed without merging**